No ocamldep link with --disable-installing-bytecode-programs

If OCaml is compiled with `--disable-installing-bytecode-programs` then the usual `ocamldep` hard-link isn't installed. This causes `ocamlfind` to fail to install:

```
#=== ERROR while compiling ocamlfind.1.8.0 ====================================#
# context     2.0.5 | linux/arm64 | ocaml-base-compiler.4.08.0 | file:///home/opam/opam-repository
# path        ~/.opam/4.08/.opam-switch/build/ocamlfind.1.8.0
# command     /usr/bin/make all
# exit-code   2
# env-file    ~/.opam/log/ocamlfind-8658-e0ba4f.env
# output-file ~/.opam/log/ocamlfind-8658-e0ba4f.out
### output ###
# [...]
# if [ "true" = "true" ]; then                 \
# 	cp topfind.ml.in topfind.ml;                             \
# else                                                             \
# 	sed -e '/PPXOPT_BEGIN/,/PPXOPT_END/ d' topfind.ml.in     \
# 		> topfind.ml ;                                   \
# fi
# ocamldep *.ml *.mli >depend
# /bin/sh: ocamldep: not found
# Makefile:181: depend: No such file or directory
# make[1]: *** [Makefile:148: depend] Error 127
# make[1]: Leaving directory '/home/opam/.opam/4.08/.opam-switch/build/ocamlfind.1.8.0/src/findlib'
# make: *** [Makefile:13: all] Error 2
```

It looks like the link step is missing from the disabled case in `tools/Makefile`:

https://github.com/ocaml/ocaml/blob/064ac56147d3bf8e7078ac2d4a96da2139be79d3/tools/Makefile#L183-L202

--- a/Changes
+++ b/Changes
@@ -260,6 +260,11 @@ Working version
   (Gabriel Scherer and Florian Angeletti,
    review by Florian Angeletti and Gabriel Radanne)
 
+- #8855, #8858: Links for tools not created when installing with
+  --disable-installing-byecode-programs (e.g. ocamldep.opt installed, but
+  ocamldep link not created)
+  (David Allsopp, report by Thomas Leonard)
+
 - #8875: fix missing newlines in the output from MSVC invocation.
   (Nicolás Ojeda Bär, review by Gabriel Scherer)
 

--- a/tools/Makefile
+++ b/tools/Makefile
@@ -197,6 +197,7 @@ else
 	do \
 	  if test -f "$$i".opt; then \
 	    $(INSTALL_PROG) "$$i.opt" "$(INSTALL_BINDIR)/$$i.opt$(EXE)"; \
+	    (cd "$(INSTALL_BINDIR)/" && $(LN) "$$i.opt$(EXE)" "$$i$(EXE)"); \
 	  fi; \
 	done
 endif

